arm9/overlays/24/src/ov24_02254CA0.c

```diff
@@ -0,0 +1,246 @@
+extern u32 ov20_02252D24();
+
+BOOL ov24_02254CA0(MemoPadDisplayHandler** displayHandlerOut, MemoPadDrawState* drawState) {
+    MemoPadDisplayHandler* displayHandler = AllocFromHeap(HEAP_ID_POKETCH_APP, sizeof(MemoPadDisplayHandler));
+    if (displayHandler) {
+        GF_ASSERT(GF_heap_c_dummy_return_true(HEAP_ID_POKETCH_MAIN));
+        ov20_022536F4(&(displayHandler->unk8), 0x10);
+        GF_ASSERT(GF_heap_c_dummy_return_true(HEAP_ID_POKETCH_MAIN));
+        displayHandler->drawState = drawState;
+        displayHandler->config = (BgConfig*)ov20_02252D34();
+        displayHandler->unk50 = ov20_02252D24();
+        GF_ASSERT(GF_heap_c_dummy_return_true(HEAP_ID_POKETCH_MAIN));
+        *displayHandlerOut = displayHandler;
+        return TRUE;
+    }
+    return FALSE;
+}
+
+BOOL ov24_02254D00(MemoPadDisplayHandler* displayHandler) {
+    static const WindowTemplate template = {
+        .bgId = GF_BG_LYR_SUB_3,
+        .left = 2,
+        .top = 2,
+        .width = 20,
+        .height = 19,
+        .palette = 0,
+        .baseTile = 12,
+    };
+    displayHandler->window = AllocWindows(HEAP_ID_POKETCH_APP, 1);
+    if (displayHandler->window) {
+        AddWindow(displayHandler->config, displayHandler->window, &template);
+        if (sub_0208946C(displayHandler->drawState->unk16E8, displayHandler->window->pixelBuffer, 0x2f80) == 0) {
+            FillWindowPixelBuffer(displayHandler->window, MEMO_PAD_PIXEL_TYPE_EMPTY);
+        }
+        return TRUE;
+    }
+    return FALSE;
+}
+
+void ov24_02254D48(MemoPadDisplayHandler* displayHandler) {
+    if (displayHandler) {
+        GF_ASSERT(GF_heap_c_dummy_return_true(HEAP_ID_POKETCH_MAIN));
+        if (displayHandler->window) {
+            sub_02089444(displayHandler->drawState->unk16E8, displayHandler->window->pixelBuffer, 0x2f80);
+            RemoveWindow(displayHandler->window);
+            FreeToHeap(displayHandler->window);
+        }
+        FreeToHeap(displayHandler);
+    }
+}
+
+void ov24_02254D8C(MemoPadDisplayHandler* displayHandler, u32 arg1) {
+    // TODO: types
+    static const u32 ov24_0225516C[] = {
+        0x00, (u32)ov24_02254DDC, 0x00,
+        0x01, (u32)ov24_02254EE0, 0x00,
+        0x02, (u32)ov24_02254F28, 0x00,
+        0x03, (u32)ov24_02254F40, 0x00,
+        0x04, (u32)ov24_02255038, 0x00,
+        0x05, (u32)ov24_02255050, 0x00,
+        0x00, 0x00000000, 0x00,
+    };
+    ov20_022537E0(ov24_0225516C, arg1, displayHandler, displayHandler->drawState, &(displayHandler->unk8), 2, 8);
+}
+
+BOOL ov24_02254DB0(MemoPadDisplayHandler* displayHandler, u8 arg1) {
+    return ov20_02253794(&(displayHandler->unk8), arg1);
+}
+
+BOOL ov24_02254DBC(MemoPadDisplayHandler* displayHandler) {
+    return ov20_022537B8(&(displayHandler->unk8));
+}
+
+void ov24_02254DC8(void* arg0) {
+    ov20_02253888(&(((MemoPadDisplayHandler*)ov20_022538A0(arg0))->unk8), arg0);
+}
+
+void ov24_02254DDC(void* arg0, void* arg1) {
+    static const BgTemplate template_7000 = {
```

Comment:
why 7000? cause of the screen base?
also should be outside the func

Reply:
yeah, naming them based on the layer they're used with might be more apt